library/Zend/Db/Sql/AbstractSql.php

```diff
@@ -108,6 +108,13 @@ protected function processExpression(ExpressionInterface $expression, PlatformIn
         $expressionParamIndex = &$this->instanceParameterIndex[$namedParameterPrefix];
 
         foreach ($parts as $part) {
+            // #7407: use $expression->getExpression() to get the unescaped version of the expression
+            if (is_string($part) && $expression instanceof Expression) {
```

Comment:
When looking at different pull requests I often see the preferred way is to use `continue` instead of `if`/`elseif`/`else` structures. Personally I find the `continue` easier to read, but it is all just a mather of preference.

For now I will leave it as it is, but if the ZF2 maintainers want the other structure it is no problem and I can change it at that moment.